csharp/src/Google.Protobuf.Test.NativeAot/Reflection/Descriptors.cs

```diff
@@ -0,0 +1,128 @@
+            Assert.AreNotEqual(0, messageTypes.Count);
+            
+            foreach (var item in messageTypes)
+            {
+                Console.WriteLine($"{item.Name} message");
+            }
+            Console.WriteLine($"{messageTypes.Count} messages");
+        }
+        
+        public static void GetMessageFields()
+        {
+            var fields = TestAllTypesProto3.Descriptor.Fields.InDeclarationOrder();
+            Assert.AreNotEqual(0, fields.Count);
+            
+            foreach (var item in fields)
+            {
+                Console.WriteLine($"{item.Name} field");
+            }
+            Console.WriteLine($"{fields.Count} fields");
+        }
+
+        public static void GetMessageEnumTypes()
+        {
+            var enumTypes = TestAllTypesProto3.Descriptor.EnumTypes;
+            Assert.AreNotEqual(0, enumTypes.Count);
+
+            foreach (var item in enumTypes)
+            {
+                Console.WriteLine($"{item.Name} enum type");
+            }
+            Console.WriteLine($"{enumTypes.Count} enum types");
+        }
+
+        public static void GetEnumFields()
+        {
+            // There isn't a descriptor for enum types. Use standard .NET reflection to check fields.
+            var enumFields = typeof(TestAllTypesProto3.Types.NestedEnum).GetFields();
+            Assert.AreNotEqual(0, enumFields.Length);
+
+            foreach (var item in enumFields)
+            {
+                Console.WriteLine($"{item.Name} enum field");
+            }
+            Console.WriteLine($"{enumFields.Length} enum fields");
+        }
+
+        public static void AccessorSingleFieldReflection()
+        {
+            var field = TestAllTypesProto3.Descriptor.FindFieldByName("fieldname1");
+            var message = (IMessage)Activator.CreateInstance(TestAllTypesProto3.Descriptor.ClrType)!;
+
+            field.Accessor.SetValue(message, 101);
+
+            var value = (int)field.Accessor.GetValue(message);
+            Assert.AreEqual(101, value);
+
+            field.Accessor.Clear(message);
+
+            value = (int) field.Accessor.GetValue(message);
+            Assert.AreEqual(0, value);
+        }
+
+        public static void AccessorOneOfFieldReflection()
+        {
+            var bytesfield = TestAllTypesProto3.Descriptor.FindFieldByName("oneof_bytes");
+            var stringfield = TestAllTypesProto3.Descriptor.FindFieldByName("oneof_string");
+            var message = (IMessage) Activator.CreateInstance(TestAllTypesProto3.Descriptor.ClrType)!;
+
+            bytesfield.Accessor.SetValue(message, ByteString.CopyFrom(new byte[] { 1, 2, 3 }));
+
+            var bytesValue = (ByteString) bytesfield.Accessor.GetValue(message);
+            if (bytesValue.Span is not [1, 2, 3])
+            {
+                throw new Exception($"Expected 1,2,3.");
+            }
+
+            if (!bytesfield.Accessor.HasValue(message))
```

Comment:
Any reason for not using Assert.AreEqual for the next three assertions? (We could add Assert.True and Assert.False if necessary too, although I'm not bothered by `Assert.AreEqual(true, xyz)`.)

Reply:
I added the slim `Assert` class after I wrote this. Can update.